src/main/java/io/cdap/delta/datastream/DatastreamEventReader.java

```diff
@@ -107,7 +113,14 @@ public DatastreamEventReader(DatastreamConfig config, EventReaderDefinition defi
     this.config = config;
     this.definition = definition;
     this.emitter = emitter;
-    this.executorService = Executors.newSingleThreadScheduledExecutor();
+
+    this.bucketCreatedByCDF = false;
+    if (context.getState(BUCKET_CREATED_BY_CDF) != null) {
```

Comment:
1. avoid calling the getState Api twice. 
2. check for empty bytes array. 

you can directly use boolean expression : 
```
byte[] state = context.getState(BUCKET_CREATED_BY_CDF);
this.bucketCreatedByCDF = (state != null && state.length != 0 && Bytes.toBoolean(state));
```